pkg/mimir/mimir.go

```diff
@@ -767,6 +761,26 @@ func New(cfg Config, reg prometheus.Registerer) (*Mimir, error) {
 	return mimir, nil
 }
 
+func setUpGoRuntimeMetrics(cfg Config, reg prometheus.Registerer) {
+	rules := []collectors.GoRuntimeMetricsRule{
+		// Enable the mutex wait time metric.
+		{Matcher: goregexp.MustCompile(`^/sync/mutex/wait/total:seconds$`)},
+	}
+
+	if cfg.EnableGoRuntimeMetrics {
+		// Enable all available runtime metrics.
+		rules = append(rules, collectors.MetricsAll)
+	}
+
+	// Unregister the default Go collector...
+	reg.Unregister(collectors.NewGoCollector())
```

Comment:
Should these lines still be inside the if statement? Same with the first line in this function.

Reply:
`EnableGoRuntimeMetrics` is a flag that enables all Go runtime metrics that is disabled by default (see https://github.com/grafana/mimir/pull/5009 for more context).

If the flag is not enabled, we want to enable the default Go runtime metrics + the mutex wait time metric.

If the flag is enabled, we want to enable everything.

Because we're not just using the default set of metrics in either scenario, we need to replace the default Go runtime metrics collector with one configured appropriately.